### PR TITLE
Add serviceMonitor for mothership-reconciler

### PR DESCRIPTION
**Pull request by vvxxvvxx**
**Description**

Changes proposed in this pull request:

- Add ServiceMonitor for mothership-reconciler.
- Add PeerAuthentication
- Allow metrics in authorization policy

**Related issue(s)**

### CODE REVIEW

File: resources/kcp/charts/mothership-reconciler/templates/service-monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{{ if .Values.global.mothership_reconciler.servicemonitor }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+{{ include "mothership-reconciler.labels" . | indent 4 }}
+  name: {{ include "mothership-reconciler.fullname" . }}
+  namespace: kcp-system
```
Comment (ebensom) [inline resources/kcp/charts/mothership-reconciler/templates/service-monitor.yaml:8]:
```suggestion
  namespace: {{ .Release.Namespace }}
```

File: resources/kcp/charts/mothership-reconciler/templates/service-monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+spec:
+  endpoints:
+  - port: http
+  namespaceSelector:
+    matchNames:
+    - kcp-system
```
Comment (ebensom) [inline resources/kcp/charts/mothership-reconciler/templates/service-monitor.yaml:14]:
```suggestion
    - {{ .Release.Namespace }}
```

**Review by ebensom — DISMISSED** [review]
In general looks LGTM, 2 small comments

**Review by suleymanakbas91 — APPROVED** [review]

**Pull request merged**